test/adjoint.jl

```diff
@@ -939,10 +1042,19 @@ _, res = adjoint_sensitivities(sol_singular_mm, alg, t = ts,
     maxiters = Int(1e6))
 reference_sol = ForwardDiff.gradient(p -> G(p, prob_singular_mm, ts,
         sol -> sum(last, sol.u)), vec(p))
-for salg in [QuadratureAdjoint(), InterpolatingAdjoint(), BacksolveAdjoint()]
+for salg in [
+    QuadratureAdjoint(),
+    InterpolatingAdjoint(),
+    BacksolveAdjoint(),
+    GaussAdjoint(),
+]
     _, res = adjoint_sensitivities(sol_singular_mm, alg, t = ts,
         dgdu_discrete = dg_singular, abstol = 1e-14,
         reltol = 1e-14, sensealg = salg,
         maxiters = Int(1e6))
-    @test res'≈reference_sol rtol=1e-7
+    if salg == GaussAdjoint()
+        @test_broken res'≈reference_sol rtol=1e-7
```

Comment:
Do we know why it's broken?

Reply:
I am not sure yet, trying to figure it out now

Reply:
Fixed the issue, GaussAdjoint passes all tests now

Reply:
To fix it, I had to add the reverse loss callback part in. I am not entirely sure what this part does. Do you know?

Reply:
Because maybe there is a smarter, more efficient way to deal with this in the IntegratingCallback

Reply:
Yes, that's the first term in https://github.com/SciML/DiffEqDevMaterials/blob/master/dae_adjoint/output/main.pdf (32)

Reply:
The code that computes this doesn't use $\lambda$ at all though

Reply:
I think I see what is happening now, and I don't think there is a more efficient way to compute this since we do not need $\lambda$ for the calculation